tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassLoaderRunner.java

```diff
@@ -121,41 +128,61 @@ public ArtifactClassLoaderRunner(Class<?> clazz, RunnerBuilder builder) throws T
      */
-    private static ClassLoaderTestRunner createClassLoaderTestRunner(Class<?> klass) throws IOException
+    private static ArtifactClassLoaderHolder createClassLoaderTestRunner(Class<?> klass) throws IOException
     {
         // Initializes utility classes
         ClassPathUrlProvider classPathUrlsProvider = new ClassPathUrlProvider();
         MavenDependenciesResolver mavenDependenciesResolver = new DependencyGraphMavenDependenciesResolver();
         MavenMultiModuleArtifactMapping mavenMultiModuleArtifactMapping = new MuleMavenMultiModuleArtifactMapping();
-        MuleClassLoaderRunnerFactory classLoaderRunnerFactory = new MuleClassLoaderRunnerFactory();
+        MuleClassLoaderIsolatedFactory classLoaderRunnerFactory = new MuleClassLoaderIsolatedFactory();
         ClassPathClassifier classPathClassifier = new MuleClassPathClassifier();
 
+        List<String> extensionBasePackages = getAnnotationAttributeFromHierarchy(klass, ArtifactClassLoaderRunnerConfig.class, "extensionBasePackage");
+        List<Class[]> exportClassesList = getAnnotationAttributeFromHierarchy(klass, ArtifactClassLoaderRunnerConfig.class, "exportClasses");
+        Set<Class> exportedClasses = exportClassesList.stream().flatMap(Arrays::stream).collect(toSet());
+
+        final File targetTestClassesFolder = new File(klass.getProtectionDomain().getCodeSource().getLocation().getPath());
+
         // Does the classification and creation of the isolated ClassLoader
-        ClassPathClassifierContext context = new ClassPathClassifierContext(klass, classPathUrlsProvider.getURLs(),
-                                                                            mavenDependenciesResolver.buildDependencies(), mavenMultiModuleArtifactMapping);
+        ClassPathClassifierContext context = new ClassPathClassifierContext(new File(targetTestClassesFolder.getParentFile(), "classes"), targetTestClassesFolder, classPathUrlsProvider.getURLs(),
+                                                                            mavenDependenciesResolver.buildDependencies(), mavenMultiModuleArtifactMapping,
+                                                                            splitCommaSeparatedAttributeValues("exclusions", klass), splitCommaSeparatedAttributeValues("extraBootPackages", klass),
+                                                                            extensionBasePackages, exportedClasses);
         ArtifactUrlClassification artifactUrlClassification = classPathClassifier.classify(context);
-        return classLoaderRunnerFactory.createClassLoaderTestRunner(context.getExtraBootPackages(), artifactUrlClassification);
+        return classLoaderRunnerFactory.createArtifactClassLoader(context.getExtraBootPackages(), artifactUrlClassification);
+    }
+
+    /**
+     * Gets the {@link List} of values for the given attribute name in the class hierarchy and split values as comma separated
```

Comment:
Try to rephrase this javadoc, is a bit confusing. Also add an extra line between method and parameters descriptions